core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -451,10 +449,31 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
     }
   }
 
+  private def getOrDefaultClient(
+    sanitizedClientId: Option[ClientQuotaEntity.ConfigEntity]
+  ): Option[ClientQuotaEntity.ConfigEntity] = {
+    if (sanitizedClientId.isEmpty)
+      None  
+    else if (sanitizedClientId.get.name() == DefaultString)
```

Comment:
in kraft, we don't use `<default>` so this check is weird. If this is used to fix test, then maybe we should revise the test